src/sage/modules/free_module_pseudomorphism.py

```diff
@@ -491,14 +497,21 @@ def __eq__(self, other):
             sage: h = V.hom(m)
             sage: g == h
             True
+
+        ::
+
+            sage: f < V
+            Traceback (most recent call last):
+            ...
+            TypeError: unsupported operand parent(s) for <: 'Set of Pseudoendomorphisms (twisted by z |--> z^7) of Vector space of dimension 2 over Finite Field in z of size 7^3' and '<class 'sage.modules.free_module.FreeModule_ambient_field_with_category'>'
         """
-        if isinstance(other, FreeModuleMorphism):
+        parent = self.parent()
+        if other.parent() is not parent:
             try:
-                other = self.parent()(other)
+                other = parent(other)
```

Comment:
I thought sage's coercion framework guarantee that the two arguments to `_richcmp_` must have same parent anyway? (`Element.__richcmp__`)